ztest/ztest.go

```diff
@@ -121,66 +169,114 @@ func Run(t *testing.T, dirname string) {
 			if err != nil {
 				t.Fatalf("%s: %s", filename, err)
 			}
-			out, errout, err := run(zq, zt.ZQL, zt.OutputFormat, zt.OutputFlags, zt.Input...)
-			if err != nil {
-				if zt.errRegex != nil {
-					if !zt.errRegex.Match([]byte(errout)) {
-						t.Fatalf("%s: error doesn't match expected error regex: %s %s", filename, zt.ErrorRE, errout)
-					}
-				} else {
-					if out != "" {
-						out = "\noutput:\n" + out
-					}
-					t.Fatalf("%s: %s%s", filename, err, out)
-				}
-			} else if zt.errRegex != nil {
-				t.Fatalf("%s: no error when expecting error regex: %s", filename, zt.ErrorRE)
-			}
-			expectedOut, oerr := zt.getOutput()
-			require.NoError(t, oerr)
-			if out != expectedOut {
-				a := expectedOut
-				b := out
+			run(t, testname, bindir, dirname, filename, zq, zt)
+		})
+	}
+}
 
-				if !utf8.ValidString(a) {
-					a = encodeHex(a)
-					b = encodeHex(b)
-				}
+type File struct {
+	// Name is the name of the file in which the test shell code runs.
+	Name string `yaml:"name"`
+	// Data, Hex, and Source represents the different ways file data can
+	// be defined for this file.  Data is a string turned into the contents
+	// of the file, Hex is hex decded, and Source is a string representing
+	// the pathname of a file the repo that is read to comprise the data.
+	Data   string `yaml:"data,omitempty"`
+	Hex    string `yaml:"hex,omitempty"`
+	Source string `yaml:"source,omitempty"`
```

Comment:
What's an example use case for needing `Source` where `Name` isn't sufficient?

Reply:
Shared test inputs, e.g., ../../test-data/in1.zng

Maybe overkill but it's in and it works.